torchchat/generate.py

```diff
@@ -328,6 +357,10 @@ def prefill(
                 x_sliced, ip_sliced = x[:, i].view(-1, 1), input_pos[i].view(-1)
                 # logging.debug(f"<sliced> x: {x_sliced}, input_pos: {ip_sliced}")
                 logits = model(x_sliced, ip_sliced)  # (x[:, i], input_pos[i])
+        elif batch is not None:
```

Comment:
I don't think this MM plays well with sequential_prefill. Can we bump this conditional up before that check and call out that seq_prefill is not tested with MM

Reply:
Legit